snow/engine/snowman/voter.go

```diff
@@ -75,24 +57,24 @@ func (v *voter) Update(ctx context.Context) {
 			zap.Stringer("result", &result),
 		)
 		if err := v.t.Consensus.RecordPoll(ctx, result); err != nil {
-			v.t.errs.Add(err)
+			return err
 		}
 	}
 
-	if v.t.errs.Errored() {
-		return
-	}
-
 	if err := v.t.VM.SetPreference(ctx, v.t.Consensus.Preference()); err != nil {
-		v.t.errs.Add(err)
-		return
+		return err
 	}
 
 	if v.t.Consensus.NumProcessing() == 0 {
 		v.t.Ctx.Log.Debug("Snowman engine can quiesce")
-		return
+		return nil
 	}
 
 	v.t.Ctx.Log.Debug("Snowman engine can't quiesce")
 	v.t.repoll(ctx)
+	return nil
+}
+
+func (v *voter) Cancel(ctx context.Context) error {
+	return v.Execute(ctx)
```

Comment:
Why is cancellation considered equivalent to execution? I don't know much about this part of the code, but I'd assume that a call to `Cancel()` should never happen? If so, I'd return an error.

Reply:
I refactored this a bit and added a comment - does that additional context answer your question?

Reply:
Yup, makes more sense now